### PR TITLE
host type UI updates to support dropdown search and multiple pages

### PR DESCRIPTION
**Pull request by rfleur01**
The #capacity-eng team built this dashboard that shows over-provisioned instances. Based on feedback from the observability team, capeng would like implement the following fixes to simplify workflow of selecting/changing instance types in Teletraan.

Introduce new UI updates for Host Types that will improve user experience and simplify the workflow for changing instance types this includes:
- Modifying the Host Types page to allow users to click "Next" and "Previous" to view more than 200 Host Types
- Addition of Network and Retired Columns to Host Type Page + Network field to Host Type creation
- Modification of Host Type selector to be search able and contain Network information on New Capacity and Cluster Configuration pages
- Greying out of retired instance types in Host Type selector

Test Plan
1. Navigate to the Cluster Configuration page 
2. Select the Host Type dropdown
3. Confirm that you can search/filter by provider name and abstract name
4. 
image
5. Confirm that retired host types are greyed out and unselectable
6. 
image
7. Confirm that the Host Types are filtered when the Arch field is changed
image
image
9. Click the help button next to Host Types and confirm that the Network and Retired columns are displayed
image
11. Repeat these steps on the Cluster creation page
12. Navigate to the clouds/hosttypes page
13. Confirm that there are columns for Network and Retired
14. 
image
15. Click Add Host Types
16. Confirm there is an field to input Network
17. 
image

### CODE REVIEW

**Review by liyaqin1 — APPROVED** [review]

File: deploy-board/deploy_board/static/js/components/clusterconfigcomponents.js [hunk 1]
```diff
@@ -388,6 +390,41 @@ Vue.component("placements-select", {
...
+    methods: {
+        updateValue: function (value) {
+            this.$emit('input', value)
+        },
+        helpClick: function () {
+            this.$emit('helpclick')
+        },
+        retiredChange: function (value) {
+            this.$emit('retiredclick', value)
+        },
```
Comment (osoriano) [inline deploy-board/deploy_board/static/js/components/clusterconfigcomponents.js:424]: Is this used? I tried searching for `retiredChange` / `retiredclick`, but no luck

Comment (rfleur01) [reply deploy-board/deploy_board/static/js/components/clusterconfigcomponents.js]: Not used. That was from an old commit. I removed it

File: deploy-board/deploy_board/templates/clusters/cluster_configuration.html [hunk 2]
```diff
@@ -764,11 +771,15 @@ <h4 class="panel-title pull-left pointer-cursor">
...
-                                            text: item.provider_name + " (" + item.core + " cores, " + item.mem + " GB, " + item.storage + ", " + item.provider + ": " + item.abstract_name + ")",
-                                            isSelected: item.id === data[0].id
+                                            text: item.provider_name + " (" + item.core + " cores, " + item.mem + " GB, " + item.storage + ", " + item.network + ", " + item.provider + ": " + item.abstract_name + ")",
+                                            isSelected: item.id === data[0],
+                                            isDisabled: item.retired
                                         }
                                     });
                             capacitySetting.selectedHostTypeValue = data[0].id;
+                            setTimeout(function() {
+                                $(".single-select-search").trigger("chosen:updated");
```
Comment (osoriano) [inline deploy-board/deploy_board/templates/clusters/cluster_configuration.html:781]: Looks like this happens in the background with 50ms delay. Is that expected? Could be worth having an explanation in a comment similar to: https://github.com/pinterest/teletraan/blob/468c0653600e158c6e6ebadf3771efb4b2e6412e/deploy-board/deploy_board/templates/clusters/cluster_configuration.html#L1024-L1029

Comment (osoriano) [reply deploy-board/deploy_board/templates/clusters/cluster_configuration.html:782]: Could `Vue.nextTick` be used to be consistent with the other code?

Comment (rfleur01) [reply deploy-board/deploy_board/templates/clusters/cluster_configuration.html:782]: Yes that seems to work. I will update it.

File: deploy-board/deploy_board/templates/configs/new_capacity_adv.html [hunk 3]
```diff
@@ -399,11 +408,15 @@ <h4 class="panel-title pull-left">Capacity</h4>
...
-                                    isSelected: item.id === data[0].id
+                                    text: item.abstract_name + " (" + item.core + " cores, " + item.mem + " GB, " + item.storage + ", " + item.network + ", " + item.provider + ": " + item.provider_name + ", " + item.network + ")",
+                                    isSelected: item.id === data[0].id,
+                                    isDisabled: item.retired
                                 }
                             });
                     capacitySetting.selectedHostTypeValue = data[0].id;
+                    setTimeout(function() {
+                        $(".single-select-search").trigger("chosen:updated");
+                    }, 50);
```
Comment (osoriano) [inline deploy-board/deploy_board/templates/configs/new_capacity_adv.html:419]: Based on the discussion above, this call could also be updated

**Review by osoriano — APPROVED** [review]

**Pull request merged**